teps/0048-task-results-without-results.md

```diff
@@ -670,6 +706,103 @@ Having default results defined in check-name-matches Task will fix the Pipeline.
 
 ## Alternatives
 
+### Specifying Default Value during Runtime
+
+Adding an optional field - `default` for a `Result` which can be specified
+during runtime, i.e., in `TaskRun` or `Pipeline` for a `Task` or in `PipelineRun` for a `Pipeline`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task
+spec:
+  results:
+  - name: merge_status
+    description: whether to rebase or squash
+    type: string
+  - name: branches
+    description: branches in the repository
+    type: array
+  - name: images
+    type: object
+    properties:
+      node:
+       type: string
+      gcloud:
+       type: string
+  steps:
+  ...
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: task
+spec:
+  taskRef:
+    name: task
+  results:
+  - name: merge_status
+    default: rebase
+  ...
+```
+
+Passing the default value for the `Result` at runtime is optional; validation
+of `TaskRun` or `Pipeline` or `PipelineRun` doesn't fail if the value isn't provided.
+
+Adding a default value to `Results` is optional; validation of a `Task` doesn't
+fail if the default value isn't provided.
+
+Adding a default value to a `Result` will guarantee that it will hold a value even
+if the `Task` fails to produce it. If a `Task` does not produce a `Result` that does not
+have a default value, then the `Task` should fail - see [tektoncd/pipeline#3497][tektoncd/pipeline#3497] for further details.
+
+Values specified in `TaskRun` or `Pipeline` should be overwritten by declaring
+`Task` itself, if `Task` declares a `Result` then it should be considered as
+the final value for that result.
+
+### Specifying Default Value at the time of Authoring the Task
```

Comment:
I think there is another alternative, which would be to define the value at the time of authoring the pipeline, but as part of the pipeline task as opposed to when the result is consumed via a variable.